### PR TITLE
TINY-10308: Fix table borders when using `table_style_by_css`

### PR DESCRIPTION
**Pull request by danoaky-tiny**
Related Ticket: TINY-10308

Description of Changes:
When settting table border to '0' or '0px', `border-width` styling is no longer used and only the `border` attribute is set to 0.
Only applies to the table properties dialog and when `table_style_with_css` is set to true.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [X] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [X] Milestone set
* [x] Docs ticket created (if applicable)

### CODE REVIEW

File: modules/tinymce/src/plugins/table/main/ts/ui/TableDialog.ts [hunk 1]
```diff
@@ -48,6 +48,7 @@ const applyDataToElement = (editor: Editor, tableElm: HTMLTableElement, data: Ta
 
   const shouldStyleWithCss = Options.shouldStyleWithCss(editor);
   const hasAdvancedTableTab = Options.hasAdvancedTableTab(editor);
+  const borderIsZero = data.border?.includes?.('0');
```
Comment (ltrouton) [inline modules/tinymce/src/plugins/table/main/ts/ui/TableDialog.ts:51]: Wouldn't this return true when `data.border` has a value of '10' or has another 0 in it?
We should be able to use `parseFloat` which should automatically strip any non-number characters like 'px'
```suggestion
  const borderIsZero = parseFloat(data.border) === 0;
```

File: modules/tinymce/src/plugins/table/main/ts/ui/TableDialog.ts [hunk 2]
```diff
@@ -62,18 +63,25 @@ const applyDataToElement = (editor: Editor, tableElm: HTMLTableElement, data: Ta
   }
 
   if (shouldStyleWithCss) {
-    styles['border-width'] = Utils.addPxSuffix(data.border);
+    if (borderIsZero) {
+      attrs.border = 0;
+      styles['border-width'] = '';
+    } else {
+      styles['border-width'] = Utils.addPxSuffix(data.border);
```
Comment (ltrouton) [inline modules/tinymce/src/plugins/table/main/ts/ui/TableDialog.ts:70]: Based on the DoD I think this is what we want to do
```suggestion
      styles['border-width'] = Utils.addPxSuffix(data.border);
      attrs.border = 1;
```

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableDialogStyleWithCssTest.ts [hunk 3]
```diff
@@ -223,6 +235,18 @@ describe('browser.tinymce.plugins.table.ui.TableCellDialogStyleWithCssTest', ()
           assertTable(editor, { cellSpacingAttr: '5', cellSpacingStyle: '' });
         }
       });
+
+      it('TINY-10308: Setting table border to 0px should remove border-width styles and set border="0"', async () => {
+        const editor = hook.editor();
+        initializeTable(editor, {});
```
Comment (ltrouton) [inline modules/tinymce/src/plugins/table/test/ts/browser/ui/TableDialogStyleWithCssTest.ts:241]: I think just having `initializeTable(editor, {});` won't add any `border-width` styles to cell so we aren't properly testing anything here

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableDialogWithoutCssTest.ts [hunk 4]
```diff
@@ -82,4 +82,16 @@ describe('browser.tinymce.plugins.table.TableDialogTest', () => {
     TableTestUtils.assertDialogValues(getExpectedData(2, '100%'), false, generalSelectors);
     await TableTestUtils.pClickDialogButton(editor, false);
   });
+
+  it('TINY-10308: Setting border to 0px should remove border-width styles and set border="0"', async () => {
+    const editor = hook.editor();
+    for (const border of [ '0', '0px' ]) {
+      editor.setContent('<table style="border-collapse: collapse;" border="1"><tbody><tr><td>&nbsp;</td></tr></tbody></table>');
```
Comment (ltrouton) [inline modules/tinymce/src/plugins/table/test/ts/browser/ui/TableDialogWithoutCssTest.ts:89]: We should probably add a `border-width` style to one of the tds here and assert it is not removed. I am not sure if that is the ideal behaviour but is what happens on v5

File: .changes/unreleased/tinymce-TINY-10308-2024-02-22.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+project: tinymce
+kind: Fixed
+body: 'When setting table border width and `table_style_with_css` is true, only the border attribute is set to 0 and border-width styling is no longer used'
```
Comment (shanmen-tiny) [inline .changes/unreleased/tinymce-TINY-10308-2024-02-22.yaml:3]:
```suggestion
body: 'When setting table border width and `table_style_by_css` is true, only the border attribute is set to 0 and border-width styling is no longer used'
```

**Review by ltrouton — APPROVED** [review]

**Review by shanmen-tiny — APPROVED** [review]

**Pull request merged**